src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetFramework.cs

```diff
@@ -30,9 +30,18 @@ private static string ResolveManagedProfilerDirectory()
 
         private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sender, ResolveEventArgs args)
         {
-            string assemblyName = new AssemblyName(args.Name).Name;
-            var path = Path.Combine(ManagedProfilerDirectory, $"{assemblyName}.dll");
+            var assemblyName = new AssemblyName(args.Name).Name;
 
+            // On .NET Framework, having a non-US locale can cause mscorlib
+            // to enter the AssemblyResolve event when searching for resources
+            // in its satellite assemblies. Exit early so we don't cause
+            // infinite recursion.
+            if (string.Equals(assemblyName, "mscorlib.resources", StringComparison.OrdinalIgnoreCase))
```

Comment:
This is the fix for .NET Framework. Since the fix is commented out, we will see a failure in CI, at which point I will re-enable the fix.

Reply:
Would it be safer to check for any resource assembly (`*.resources`)?

Reply:
No other resource assembly will cause this recursion issue so I think it's helpful to single out this particular issue. We could add another check for resources assemblies in general as a small optimization if you'd like, but it wouldn't necessarily provide any more safety.

Reply:
Should we not limit our checks to only our dependencies?

Reply:
In other words, wouldn't this be resolved by short-circuiting when the assembly name is not one of those *WE* need?